### DISCUSSION TITLE
What are the best practices for filtering by a non-required field in Prisma?

### DISCUSSION THREAD

**Discussion by zbarakzai** [prisma/prisma#13103] [Q&A]
Consider developing an e-commerce platform using Prisma and implementing a Product filter, as illustrated in the example below. However, the code below has too many if checks, and I believe there should be a simple approach to make these sorts of filters generic so that they may be utilized for other models as well.

filters input came from Graphql is something like this --> [{field: 'query', value: 'some value'}, {field: 'brand', value: '89787ecc88s00ww'}]

```
let andWhere: Prisma.ProductWhereInput = {}
let orderByQuery: Prisma.ProductOrderByWithRelationInput = {}

filters?.forEach((filter) => {

        if (
          filter.field === 'brand' ||
          filter.field === 'supplier' ||
          filter.field === 'productType'
        ) {
          andWhere[filter.field] = { id: filter.value }
        }

        if (filter.field === 'isActive') {
          andWhere.active = filter.value === 'true' ? true : false
        }

        if (filter.field === 'tags') {
          andWhere.tags = { some: { id: { equals: filter.value } } }
        }

        if (filter.field === 'query') {
          andWhere.OR = ['name', 'brand', 'productType', 'supplier']
            .map(field => {
              if (field === 'name') {
                return { name: { contains: filter.value as string, mode: 'insensitive' } }
              }

              return { [field]: { name: { contains: filter.value, mode: 'insensitive' } } }
            })
        }
      })

      if (
        orderBy?.field === 'brand' ||
        orderBy?.field === 'supplier' ||
        orderBy?.field === 'productType'
      ) {
        orderByQuery[orderBy.field] = { name: order }
      }

      if (orderBy?.field === 'name') {
        orderByQuery.name = order
      }

      if (orderBy?.field === 'created_at') {
        orderByQuery.createdAt = order
      }
```

**Comment by nurul3101** [comment]
Hey @zbarakzai 👋 

 These conditions would be dependent on your business logic so you should be able to simplify these conditions by using something like [json-rules-engine](npmjs.com/package/json-rules-engine) and making it generic so that it can be utilised by other models.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
